### DISCUSSION TITLE
Detect difference between populated input and non-populated input for uncontrolled form

### DISCUSSION THREAD

**Discussion by EliLauff** [react-hook-form/react-hook-form#9854] [Q&A]
Hi, I have a scenario where I'm attempting to distinguish between an empty input (with innerText === `''`) and a populated input (innerText !== `''`) for the purposes of displaying a placeholder.

The two inputs are being controlled by RHF via `...register('<fieldname>')` and are otherwise uncontrolled (i.e. do not have `value` passed in as a prop).  

Currently the inputs are reporting `''` as their `innerText` value regardless of `defaultValues` applied or values typed into the inputs.   Does anyone have any insight into how RHF is managing state here?  How is it that it is populating the inputs via `defaultValue` but we're still unable to detect a value there, even via DOM manipulation? 

We would like to avoid using `<Controller/>` if possible.  We're aware that grabbing the `value` from `<Controller/>` and passing it into the input is a possible solution.

**Comment by Moshyfawn** [comment]
You can [`watch`](https://react-hook-form.com/api/useform/watch) / [`useWatch`](https://react-hook-form.com/api/usewatch) the field's value in question _(or multiple fields)_ and display the placeholder depending on the value.

DOM attributes aren't reactive with React. You won't get a UI re-render and placeholder set if `innerText` is updated.
